### PR TITLE
follow #8463 #14157 and document cstring literals modification is not allowed

### PR DESCRIPTION
**Pull request by ringabout**

### CODE REVIEW

**Comment by juancarlospaco** [comment]
Looks Ok to me. Maybe that should be mentioned on https://nim-lang.github.io/Nim/cstrutils.html too. :thinking:

File: doc/manual.rst [hunk 1]
```diff
@@ -1282,6 +1282,19 @@ string from a cstring:
   var cstr: cstring = str
   var newstr: string = $cstr
 
+In C backend You shouldn't modify cstring literals. Otherwise your program will crash.
```
Comment (Araq) [inline doc/manual.rst]: This is our manual. We don't use "You" here. "will crash" is also too informal language.

File: doc/manual.rst [hunk 2]
```diff
@@ -1282,6 +1282,19 @@ string from a cstring:
...
+
+.. code-block:: nim
+  var x = cstring"literals"
+  x[1] = 'A' # This is wrong!!!
+
+cstring could be modified if it originates from a regular memory(not read-only memory):
```
Comment (Araq) [inline doc/manual.rst]: Spacing, casing. This is a *manual*.

File: doc/manual.rst [hunk 3]
```diff
@@ -1282,6 +1282,21 @@ string from a cstring:
...
   var newstr: string = $cstr
 
+For the C backend, ``cstring`` literals shouldn't be modified.
+Otherwise it will cause segmentation fault:
+
+.. code-block:: nim
+  var x = cstring"literals"
+  x[1] = 'A' # This is wrong!!!
+
+If ``cstring`` originates from a regular memory(not read-only memory),
```
Comment (juancarlospaco) [inline doc/manual.rst]:
```suggestion
If the ``cstring`` originates from a regular memory (not read-only memory),
```

File: doc/manual.rst [hunk 4]
```diff
@@ -1282,6 +1282,21 @@ string from a cstring:
   var cstr: cstring = str
   var newstr: string = $cstr
 
+For the C backend, ``cstring`` literals shouldn't be modified.
```
Comment (juancarlospaco) [inline doc/manual.rst]: And C++ backend ?.

Comment (ringabout) [inline doc/manual.rst]:
```suggestion
For the C/C++ backend, ``cstring`` literals shouldn't be modified.
```

File: doc/manual.rst [hunk 5]
```diff
@@ -1282,6 +1282,21 @@ string from a cstring:
   var cstr: cstring = str
   var newstr: string = $cstr
 
+For the C/C++ backend, ``cstring`` literals shouldn't be modified.
```
Comment (Araq) [inline doc/manual.rst]:
```suggestion
``cstring`` literals shouldn't be modified.
```

File: doc/manual.rst [hunk 6]
```diff
@@ -1282,6 +1282,21 @@ string from a cstring:
...
+Otherwise it will cause segmentation fault:
```
Comment (Araq) [inline doc/manual.rst]:
```suggestion
```

File: doc/manual.rst [hunk 7]
```diff
@@ -1282,6 +1282,20 @@ string from a cstring:
...
   var newstr: string = $cstr
 
+``cstring`` literals shouldn't be modified.
+
+.. code-block:: nim
+  var x = cstring"literals"
+  x[1] = 'A' # This is wrong!!!
+
+If the ``cstring`` originates from a regular memory (not read-only memory),
+it could be modified:
```
Comment (Araq) [inline doc/manual.rst]:
```suggestion
it can be modified:
```

**Pull request merged**